thewalrus/quantum/__init__.py

```diff
@@ -0,0 +1,202 @@
+# Copyright 2019-2020 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Quantum algorithms
+==================
+
+.. currentmodule:: thewalrus.quantum
+
+This submodule provides access to various utility functions that act on Gaussian
+quantum states.
+
+For more details on how the hafnian relates to various properties of Gaussian quantum
+states, see:
+
+* Kruse, R., Hamilton, C. S., Sansoni, L., Barkhofen, S., Silberhorn, C., & Jex, I.
+  "Detailed study of Gaussian boson sampling." `Physical Review A 100, 032326 (2019)
+  <https://journals.aps.org/pra/abstract/10.1103/PhysRevA.100.032326>`_
+
+* Hamilton, C. S., Kruse, R., Sansoni, L., Barkhofen, S., Silberhorn, C., & Jex, I.
+  "Gaussian boson sampling." `Physical Review Letters, 119(17), 170501 (2017)
+  <https://journals.aps.org/prl/abstract/10.1103/PhysRevLett.119.170501>`_
+
+* Quesada, N.
+  "Franck-Condon factors by counting perfect matchings of graphs with loops."
+  `Journal of Chemical Physics 150, 164113 (2019) <https://aip.scitation.org/doi/10.1063/1.5086387>`_
+
+* Quesada, N., Helt, L. G., Izaac, J., Arrazola, J. M., Shahrokhshahi, R., Myers, C. R., & Sabapathy, K. K.
+  "Simulating realistic non-Gaussian state preparation." `Physical Review A 100, 022341 (2019)
+  <https://journals.aps.org/pra/abstract/10.1103/PhysRevA.100.022341>`_
+
+
+Fock states and tensors
+-----------------------
+
+.. autosummary::
+
+    pure_state_amplitude
+    state_vector
+    density_matrix_element
+    density_matrix
+    fock_tensor
+    probabilities
+    update_probabilities_with_loss
+    update_probabilities_with_noise
+    normal_ordered_expectation
+    fidelity
+
+Details
+^^^^^^^
+
+.. autofunction::
+    pure_state_amplitude
+
+.. autofunction::
+    state_vector
+
+.. autofunction::
+    density_matrix_element
+
+.. autofunction::
+    density_matrix
+
+.. autofunction::
+    fock_tensor
+
+.. autofunction::
+    probabilities
+
+.. autofunction::
+    update_probabilities_with_loss
+
+.. autofunction::
+    update_probabilities_with_noise
+
+.. autofunction::
+    normal_ordered_expectation
+
+.. autofunction::
+    fidelity
+
+
+Utility functions
+-----------------
+
+.. autosummary::
```

Comment:
@thisac, it probably makes sense to regroup these based on the refactor groupings? Curious to hear what @nquesada thinks, but if the grouping makes it cleaner for developers, it might make it easier for users to navigate

Reply:
This sounds like a good idea, it is nicer to have smaller sized tables than one that is massive and hard to digest.

Reply:
I agree. :+1: